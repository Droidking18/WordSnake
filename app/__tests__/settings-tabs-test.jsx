import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent } from '@testing-library/react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';
import Settings from '../(tabs)/settings';

jest.mock('@expo/vector-icons/Ionicons', () => 'Ionicons');
jest.mock('react-native/Libraries/Image/Image', () => 'Image');
jest.mock('@/components/Collapsible', () => ({ Collapsible: 'Collapsible' }));
jest.mock('@/components/ExternalLink', () => 'ExternalLink');
jest.mock('@/components/ParallaxScrollView', () => 'ParallaxScrollView');
jest.mock('@/components/ThemedTextInput', () => ({ ThemedTextInput: 'ThemedTextInput' }));
jest.mock('@/components/ThemedText', () => ({ ThemedText: 'ThemedText' }));
jest.mock('@/components/ThemedView', () => ({ ThemedView: 'ThemedView' }));

jest.mock('@react-native-async-storage/async-storage', () => ({
  getItem: jest.fn(),
  setItem: jest.fn(),
}));

describe('Settings Screen', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Settings />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains the collapsible sections', () => {
    const tree = renderer.create(<Settings />);
    const instance = tree.root;
    const collapsibleSections = instance.findAllByType('Collapsible');
    expect(collapsibleSections.length).toBe(5); // Adjust based on actual number of collapsible sections
  });

  it('contains input fields for speed, letter generation amount, and min word length', () => {
    const tree = renderer.create(<Settings />);
    const instance = tree.root;
    const inputs = instance.findAllByType('ThemedTextInput');
    expect(inputs.length).toBe(3); // Check if there are 3 input fields
  });

  it('contains switches for wrapping and touch controls', () => {
    const tree = renderer.create(<Settings />);
    const instance = tree.root;
    const switches = instance.findAllByType('Switch');
    expect(switches).toBeTruthy(); // Check if there are 2 switches
  });

  it('contains save button', () => {
    const tree = renderer.create(<Settings />);
    const instance = tree.root;
    const saveButton = instance.findAllByProps({ style: { backgroundColor: 'white' } });
    expect(saveButton).toBeTruthy();
  });
});
