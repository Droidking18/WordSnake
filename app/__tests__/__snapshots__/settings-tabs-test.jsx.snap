// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Screen renders correctly 1`] = `
<ThemedView
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "paddingTop": 50,
    }
  }
>
  <ThemedView
    style={
      {
        "flexDirection": "row",
        "gap": 8,
      }
    }
  >
    <ThemedText
      type="title"
    >
      Settings
    </ThemedText>
  </ThemedView>
  <ThemedView
    contentContainerStyle={
      {
        "alignItems": "center",
        "justifyContent": "flex-start",
      }
    }
    scrollable={true}
    style={
      {
        "flex": 1,
        "width": "100%",
      }
    }
  >
    <ThemedText
      style={
        {
          "fontWeight": "bold",
          "paddingBottom": 20,
          "textAlign": "center",
        }
      }
    >
      Configure your playing experience!
    </ThemedText>
    <Collapsible
      title="Select the speed of your snake"
    >
      <ThemedText
        style={
          {
            "paddingBottom": 20,
            "textAlign": "center",
          }
        }
      >
        100 is default speed, 200 is double speed, 50 is half speed. The faster the snake, the more points you get per word. 200 would be double points, but it will also be more difficult to control the snake.
      </ThemedText>
    </Collapsible>
    <ThemedText
      style={
        {
          "paddingBottom": 20,
          "textAlign": "center",
        }
      }
    >
      Currently 
    </ThemedText>
    <ThemedTextInput
      onChangeText={[Function]}
      placeholder="Select a speed"
      style={
        {
          "borderRadius": 20,
          "borderWidth": 1,
          "height": 40,
          "marginBottom": 20,
          "textAlign": "center",
          "width": 600,
        }
      }
      value={100}
    />
    <Collapsible
      title="Select the amount of letters to generate"
    >
      <ThemedText
        style={
          {
            "paddingBottom": 20,
            "textAlign": "center",
          }
        }
      >
        3 is default amount. 1 is minimum amount, 6 is maximum amount. This is the amount of letters that will be generated at a time. The less letters, the more points you get per word, but it will also be more difficult to form words.
      </ThemedText>
    </Collapsible>
    <ThemedText
      style={
        {
          "paddingBottom": 20,
          "textAlign": "center",
        }
      }
    >
      Currently 
    </ThemedText>
    <ThemedTextInput
      onChangeText={[Function]}
      placeholder="Select a letter generation amount"
      style={
        {
          "borderRadius": 20,
          "borderWidth": 1,
          "height": 40,
          "marginBottom": 20,
          "textAlign": "center",
          "width": 600,
        }
      }
      value={3}
    />
    <Collapsible
      title="Select the minimum word length"
    >
      <ThemedText
        style={
          {
            "paddingBottom": 20,
            "textAlign": "center",
          }
        }
      >
        The default is 1. The minimum is 1, the maximum is 6. This is the minimum amount of letters that a word must contain to be valid. The higher the minimum letters, the more points you get per word, but it will also be more difficult to form words.
      </ThemedText>
    </Collapsible>
    <ThemedText
      style={
        {
          "paddingBottom": 20,
          "textAlign": "center",
        }
      }
    >
      Currently 
    </ThemedText>
    <ThemedTextInput
      onChangeText={[Function]}
      placeholder="Select a min word length"
      style={
        {
          "borderRadius": 20,
          "borderWidth": 1,
          "height": 40,
          "marginBottom": 20,
          "textAlign": "center",
          "width": 600,
        }
      }
      value={1}
    />
    <Collapsible
      title="Allow the snake to wrap around the screen"
    >
      <ThemedText
        style={
          {
            "paddingBottom": 20,
            "textAlign": "center",
          }
        }
      >
        If enabled, the snake will be able to wrap around the screen. This means that if the snake goes off the screen on the left, it will appear on the right. If the snake goes off the screen on the top, it will appear on the bottom. If disabled, the snake will die if it goes off the screen. Enabling this will lower the amount of points you get per word.
      </ThemedText>
    </Collapsible>
    <ThemedText
      style={
        {
          "paddingBottom": 20,
          "textAlign": "center",
        }
      }
    >
      Currently 
      Disabled
    </ThemedText>
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "height": 31,
          "width": 51,
        }
      }
      value={true}
    />
    <Collapsible
      title="Enable touch controls"
    >
      <ThemedText
        style={
          {
            "paddingBottom": 20,
            "textAlign": "center",
          }
        }
      >
        If enabled, you can control the snake by pressing buttons at the bottom of the screen. If disabled, you can control the snake by swiping on the screen. This will not affect the scoring system.
      </ThemedText>
    </Collapsible>
    <ThemedText
      style={
        {
          "paddingBottom": 20,
          "textAlign": "center",
        }
      }
    >
      Currently 
      Disabled
    </ThemedText>
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "height": 31,
          "width": 51,
        }
      }
      value={true}
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "black",
          "borderRadius": 20,
          "marginTop": 10,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Ionicons
        color="white"
        name="save"
        size={24}
        style={
          {
            "textAlign": "center",
          }
        }
      />
      <ThemedText
        style={
          {
            "color": "white",
          }
        }
      >
        Save settings
      </ThemedText>
    </View>
  </ThemedView>
</ThemedView>
`;
